woocommerce-services.php

```diff
@@ -819,6 +822,74 @@ public function deactivate_notice() {
 			<?php
 		}
 
+		public function check_jetpack_install() {
+			if ( defined( 'JETPACK_DEV_DEBUG' ) ) {
+				return true;
+			}
+
+			if ( ! class_exists( 'Jetpack_Data' ) ) {
+				return false;
+			}
+
+			$user_token = Jetpack_Data::get_access_token( JETPACK_MASTER_USER );
+			return $user_token && is_object( $user_token ) && isset( $user_token->external_user_id );
+		}
+
+		public function admin_jetpack_notice() {
+			wp_enqueue_style( 'wc_connect_banner' ); // Including the CSS this early will mostly prevent a flash of Core styles in the banner
+			add_action( 'admin_notices', array( $this, 'show_jetpack_notice' ) );
+		}
+
+		public function show_jetpack_notice() {
+			$button_onclick = '';
+			$button_url = '#';
+
+			if ( class_exists( 'Jetpack_Data' ) ) {
+				$screen = get_current_screen();
+				if ( 'jetpack' === $screen->parent_base || 'plugins' === $screen->base ) {
+					return; // Jetpack Dashboard and the plugins list screen already have big "Connect to WordPress.com" buttons
+				}
+
+				$notice_text = __( 'To get started please connect Jetpack to your Worpress.com account.', 'woocommerce-services' );
+				$button_label = __( 'Connect to WordPress.com', 'woocommerce-services' );
+				$redirect_to = admin_url( 'admin.php?page=wc-settings&tab=shipping' );
+				$button_url = Jetpack::init()->build_connect_url( false, $redirect_to, 'woocommerce-services' );
+			} else {
+				$activate_url = wp_nonce_url( 'plugins.php?action=activate&plugin=jetpack/jetpack.php', 'activate-plugin_jetpack/jetpack.php' );
+
+				if ( 0 === validate_plugin( 'jetpack/jetpack.php' ) ) {
+					$notice_text = __( 'To get started you need to activate Jetpack.', 'woocommerce-services' );
+					$button_label = __( 'Activate Jetpack', 'woocommerce-services' );
+					$button_url = $activate_url;
+				} else {
+					$notice_text = __( 'To get started you need to install Jetpack.', 'woocommerce-services' );
+					$button_label = __( 'Install Jetpack', 'woocommerce-services' );
+
+					wp_enqueue_script( 'updates' );
+					$button_progress_label = esc_js( __( 'Installing...', 'woocommerce-services' ) );
+					$error_message = esc_js( __( 'There was an error installing Jetpack. Please try installing it manually.', 'woocommerce-services' ) );
+					$button_onclick = "this.onclick = null;" .
+						"this.innerHTML = '$button_progress_label';" .
+						"this.className += ' disabled';" .
+						"wp.updates.installPlugin( { slug: 'jetpack' } )" .
+							".then( () => window.location = '$activate_url' )" .
+							".fail( () => { this.parentNode.className = 'error'; this.parentNode.innerHTML = '$error_message'; } )";
```

Comment:
> Rather than output this as an onclick attribute through PHP, let's put this in it's own JS file (perhaps woocommerce-services-banner.js).

Yep, this was too hacky.

> The initially referenced code that inspired the use of wp.updates turned the button from "install" to "activate" after the operation was successful (without a page refresh IIRC). Is that something we can do as well?

After the plugin is installed via AJAX, it's automatically activated, by redirecting the page to the `plugins.php?action=activate&plugin=jetpack/jetpack.php` URL. So, the experience is that you click the `Install` button, and after a while the page refreshes and you get a big Jetpack NUX splash screen.